infra/modules/service/access_logs.tf

```diff
@@ -0,0 +1,106 @@
+locals {
+  elb_account_map = {
+    "us-east-1" : "127311923021",
+    "us-east-2" : "033677994240",
+    "us-west-1" : "027434742980",
+    "us-west-2" : "797873946194"
+  }
+  log_file_transition = {
+    STANDARD_IA = 30
+    GLACIER     = 60
+  }
+
+  log_file_deletion = 0
+}
+
+resource "aws_s3_bucket" "access_logs" {
+  bucket_prefix = "${var.service_name}-access-logs"
+  force_destroy = false
+  # checkov:skip=CKV2_AWS_62:Event notification not necessary for this bucket expecially due to likely use of lifecycle rules
+  # checkov:skip=CKV_AWS_18:Access logging was not considered necessary for this bucket
+  # checkov:skip=CKV_AWS_144:Not considered critical to the point of cross region replication
+  # checkov:skip=CKV_AWS_300:Known issue where Checkov gets confused by multiple rules
+  # 
+}
+
+resource "aws_s3_bucket_public_access_block" "access_logs" {
+  bucket = aws_s3_bucket.access_logs.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+data "aws_iam_policy_document" "load_balancer_logs_put_access" {
+  statement {
+    effect = "Allow"
+    resources = [
+      aws_s3_bucket.access_logs.arn,
+      "${aws_s3_bucket.access_logs.arn}/*"
+    ]
+    actions = ["s3:PutObject"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${local.elb_account_map[data.aws_region.current.name]}:root"]
+    }
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "lc" {
+  count  = local.log_file_transition != [] && local.log_file_deletion != 0 ? 1 : 0
+  bucket = aws_s3_bucket.access_logs.id
+  rule {
+    id     = "StorageClass"
+    status = "Enabled"
+    dynamic "transition" {
+      for_each = local.log_file_transition
+      content {
+        days          = transition.value
+        storage_class = transition.key
+      }
+    }
+  }
+  rule {
+    id     = "AbortIncompleteUpload"
+    status = "Enabled"
+    abort_incomplete_multipart_upload {
+      days_after_initiation = 7
+    }
+  }
+  dynamic "rule" {
+    for_each = local.log_file_deletion != 0 ? [1] : []
+    content {
+      id     = "Expiration"
+      status = "Enabled"
+      expiration {
+        days = local.log_file_deletion
+      }
+    }
+  }
```

Comment:
let's get rid of this block to simplify the code, i don't see the need to support deletion by default. projects can add if they want
```suggestion
```

Reply:
I don't quite see the need to do this. This is removing flexibility from the module while not removing much in terms of code. Additionally, access logs are one of the files that can grow quickly in terms of storage and be considered low priority in terms of retention. Expiration seems like a useful thing to have prepared in the template

Reply:
i want to optimize for simplicity, avoid dynamic logic as much as possible, and minimize the number of decisions a project team has to make. if we want to keep this, is suggest having a static hardcoded rule set to 7 year retention. otherwise let's get rid of it.

Reply:
@charles-nava just checking if this comment got lost in the shuffle (i think i accidentally added it as a single comment rather than as part of the review) – could we do a quick follow up PR to make this a static block (e.g. 7 year retention)?